dev/com.ibm.ws.jdbc_fat_derby/test-applications/jdbcapp/src/jdbc/fat/derby/web/JDBCDerbyServlet.java

```diff
@@ -251,4 +333,124 @@ public void testResourceRefBehavior() throws Throwable {
+     *
+     * Test with and without a resource reference that conn.getTransactionIsolation() returns
+     * Driver's isolation level when data source definition is set to an isolation level of TRANSACTION_NONE.
+     */
+    @Test
+    public void testTNResRefBehaviorDSDef() throws Throwable {
+        Connection con = null;
+        int expected = Connection.TRANSACTION_NONE;
+
+        try {
+            con = ds6.getConnection();
+            assertEquals("Connection with dsConfig = TRAN_NONE and no res-ref should use driver default iso lvl: ", expected, con.getTransactionIsolation());
+        } finally {
+            con.close();
+        }
+
+        try {
+            con = ds7.getConnection();
+            assertEquals("Connection with dsConfig = TRAN_NONE and res-ref = TRAN_NONE should use driver default iso lvl: ", expected, con.getTransactionIsolation());
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Data Source - ds8 = DataSourceDef(no iso lvl) + Res-ref (TRAN_NONE)
+     * Data Source - ds9 = DataSourceDef(TRAN_SERIALIZABLE) + Res-ref (TRAN_NONE)
+     *
+     * Test that when resource reference isolation level is TRANSACTION_NONE that our behavior is unchanged if the
+     * data source definition isolation level is unspecified or is something other than TRANSACTION_NONE
+     */
+    @Test
+    public void testTNResRefNoneBehaviorDSDef() throws Throwable {
+        Connection con = null;
+        try {
+            int expected = Connection.TRANSACTION_REPEATABLE_READ; //WAS default
+            con = ds8.getConnection();
+            assertEquals("Connection with no dsConfig and res-ref = TRAN_NONE should use WAS default iso lvl: ", expected, con.getTransactionIsolation());
+        } finally {
+            con.close();
+        }
+
+        try {
+            int expected = Connection.TRANSACTION_SERIALIZABLE; //dsConfig
+            con = ds9.getConnection();
+            assertEquals("Connection with dsConfig = TRAN_SERIALIZABLE and res-ref = TRAN_NONE should use dsConfig iso lvl: ", expected, con.getTransactionIsolation());
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Called by testTNConfigIsoLvl in JDBCDerbyTest
+     * Data Source - ds11 = DSConfig (TRAN_NONE) + No Res-ref + TRAN_NONE JDBC Driver
+     *
+     * Ensure that the unmodified data source has the expected isolation level.
+     */
+    public void testTNOriginalIsoLvl() throws Throwable {
+        int expected = Connection.TRANSACTION_NONE;
+        final Connection con = ds11.getConnection();
+        try {
+            int actual = con.getTransactionIsolation();
+            assertEquals("Connection should have had an isolation level of: ", expected, actual);
+        } finally {
+            con.close();
+        }
+    }
+
+    /**
+     * Called by testTNConfigIsoLvl in JDBCDerbyTest
+     * Data Source - ds11 = DSConfig (TRAN_NONE) + No Res-ref + TRAN_NONE JDBC Driver
```

Comment:
General comment, it feels like we have a lot of duplicated functionality, are we able to consolidate some of this?